pandas/io/tests/test_parsers.py

```diff
@@ -3070,6 +3070,26 @@ def test_whitespace_lines(self):
         df = self.read_csv(StringIO(data))
         tm.assert_almost_equal(df.values, expected)
 
+    @tm.network
```

Comment:
With `compression="infer"`, we look at the `Content-Encoding` header from the response.  There's no way to get `raw.github` to reply with a `Content-Encoding` header, so I had to use an s3 url, where you can set arbitrary response headers.